components/warn_metric_ack_modal/warn_metric_ack_modal.tsx

```diff
@@ -0,0 +1,242 @@
+                defaultMessage='Acknowledge'
+            />
+        );
+        const subText = (
+            <div
+                style={{opacity: '0.45'}}
+                className='help__format-text'
+            >
+                <FormattedMarkdownMessage
+                    id='warn_metric_ack_modal.mailto.body.sub_text'
+                    defaultMessage='Acknowledgement will be sent to Mattermost, Inc.'
+                />
+            </div>
+        );
+        return (
+            <Modal
+                dialogClassName='a11y__modal'
+                show={this.props.show}
+                keyboard={false}
+                onHide={this.onHide}
+                onExited={this.onHide}
+                role='dialog'
+                aria-labelledby='warnMetricAckHeaderModalLabel'
+            >
+                <Modal.Header closeButton={true}>
+                    <Modal.Title
+                        componentClass='h1'
+                        id='warnMetricAckHeaderModalLabel'
+                    >
+                        {headerTitle}
+                    </Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    <div>
+                        {descriptionText}
+                        <br/>
+                        {this.renderError()}
+                        <br/>
+                        {subText}
+                    </div>
+                </Modal.Body>
+                <Modal.Footer>
+                    <button
+                        type='button'
+                        className='btn btn-link cancel-button'
+                        onClick={this.onHide}
+                    >
+                        <FormattedMessage
+                            id='warn_metric_ack_modal.cancel'
+                            defaultMessage='Cancel'
+                        />
+                    </button>
+                    <button
+                        type='button'
+                        className='btn btn-primary save-button'
+                        data-dismiss='modal'
+                        disabled={this.state.saving}
+                        autoFocus={true}
+                        onClick={this.onEmailUsClick}
+                    >
+                        {buttonText}
+                    </button>
+                </Modal.Footer>
+            </Modal>
+        );
+    }
+}
+
+type ErrorLinkProps = {
+    defaultMessage: string;
+    messageId: string;
+    onClickHandler: () => Promise<void>;
+    url: string;
+    warnMetricId: string;
+}
+
+const WarnMetricAckErrorLink: React.FC<ErrorLinkProps> = ({defaultMessage, messageId, onClickHandler, url, warnMetricId}: ErrorLinkProps) => {
+    return (
+        <a
+            href={url}
+            rel='noopener noreferrer'
+            target='_blank'
+            onClick={
+                () => {
+                    trackEvent('admin', 'click_warn_metric_ack_acknowledge', {metric: warnMetricId});
```

Comment:
I think this is send here, and in the `onClickHandler` passed here when there is an error.

Reply:
yes, it duplicated, will remove.